credsweeper/rules/rule.py

```diff
@@ -80,8 +79,7 @@ def __init__(self, config: Config, rule_dict: Dict) -> None:
             self._malformed_rule_error(rule_dict, Rule.REQUIRED_REGEX)
         self.__required_regex = re.compile(required_regex) if required_regex else None
         self.__min_line_len = int(rule_dict.get(Rule.MIN_LINE_LEN, MAX_LINE_LENGTH))
-        self.__doc_available: bool = rule_dict.get(Rule.DOC_AVAILABLE, True)
-        self.__doc_only: bool = rule_dict.get(Rule.DOC_ONLY, False)
+        self.__target: List[str] = rule_dict.get(Rule.TARGET, [])
```

Comment:
Consider using Enum for the target, to limit variants and simplify comparisons.

Reply:
Good suggestion. I'll consider it next PR.
Thank you for the review.